streams/src/main/java/org/apache/kafka/streams/processor/TaskMetadata.java

```diff
@@ -40,6 +40,18 @@
 
     private final Optional<Long> timeCurrentIdlingStarted;
 
+    /**
+     * @deprecated since 3.0, please use the constructor that accepts a TaskId object instead of a String
```

Comment:
nit: It should be more clear to put a constructor link here. ex:
```
@deprecated since 3.0, please use {@link #TaskMetadata(TaskId, Set, Map, Map, Optional)} instead
```